File: driver/languages/csharp/template.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;

public class Driver {
    private static string __phase = "init";

    public static void Main(string[] args) {
        try {
            Scanner scanner = new Scanner();
            int numTests = scanner.NextInt();

            for (int t = 0; t < numTests; t++) {
                __phase = "parse_inputs";
                {{MAIN_LOOP}}
            }
        } catch (Exception e) {
            Console.Error.WriteLine($"@@ERROR@@:{__phase}:{e.Message}");
            Environment.Exit(1);
        }
    }
}

// {{PART_SCANNER}}

// {{PART_DATA_STRUCTURES}}

// {{PART_COMPARATOR}}

{{USER_CODE}}

File: driver/languages/javascript/template.js
"use strict";

const fs = require('fs');

// {{PART_SCANNER}}

// {{PART_DATA_STRUCTURES}}

// {{PART_COMPARATOR}}

// ---- USER SOLUTION ----
{{USER_CODE}}
// ---- END USER SOLUTION ----

function main() {
    const sc = new Scanner();
    let __phase = "init";

    try {
        const num_cases = sc.nextInt();
        for (let i = 0; i < num_cases; i++) {
            __phase = "parse_inputs";
            {{MAIN_LOOP}}
        }
    } catch (e) {
        process.stderr.write(`@@ERROR@@:${__phase}:${e.message}\n`);
        process.exit(1);
    }
}

main();
